api/client/client.go

```diff
@@ -454,6 +455,7 @@ func (c *Client) dialGRPC(ctx context.Context, addr string) error {
 			otelgrpc.UnaryClientInterceptor(),
 			metadata.UnaryClientInterceptor,
 			interceptors.GRPCClientUnaryErrorInterceptor,
+			interceptors.RetryWithMFAUnaryInterceptor(c.performMFACeremony),
```

Comment:
I'd say yes, why would we not want it?

Reply:
Oh sorry, I was thinking about the wrong PR. Ignore me.

Reply:
I've changed this so it is only used for `tsh`, `tctl`, and Teleport Connect.

The interceptor will always be set, but it'll essentially be a noop unless `client.PromptAdminRequestMFA` is set.

To be specific, it will aggregate `trace.BadParameter("missing PromptAdminRequestMFA field, client cannot perform MFA ceremony")` with the `ErrAdminActionMFARequired` passed through the interceptor.

@codingllama you mentioned before that you weren't sold on enabling/disabling the interceptor in this way, but I think it makes more sense. If the client has a method to prompt MFA, it should. If it doesn't, it shouldn't. Adding a separate flag seems redundant.

Reply:
How difficult would it be to create certain clients without the interceptor?

If you think it isn't worth it, or a worse solution, we can go with the nil. Thanks for calling attention to it explicitly.

Reply:
I think having the aggregated error is slightly better as it can help point us in the right direction with any bugs, such as if a client does not have `PromptAdminRequestMFA` set when we expect it to. Also adding the mfa-retry interceptor into the middle of list after the fact is a bit awkward from a code formatting perspective.